proxy/http/HttpSM.cc

```diff
@@ -2771,7 +2771,8 @@ HttpSM::tunnel_handler_post(int event, void *data)
   }
 
   if (event != HTTP_TUNNEL_EVENT_DONE) {
-    if ((event == VC_EVENT_WRITE_COMPLETE) || (event == VC_EVENT_EOS)) {
+    if ((event == VC_EVENT_WRITE_COMPLETE) || (event == VC_EVENT_EOS) || (event == VC_EVENT_ERROR) ||
```

Comment:
I think we should add the inline function to make sure that the same condition is used under the same context, and to clarify the condition we expect.

The switch case style is more readable than current style though, it doesn't tell us what the case is.